extensions-core/core/runtime/src/main/java/org/apache/camel/quarkus/core/RuntimeBeanRepository.java

```diff
@@ -80,6 +81,17 @@ private static <T> Set<T> getReferencesByType(BeanManager manager, Class<T> type
     private static <T> Optional<T> getReferenceByName(BeanManager manager, String name, Class<T> type) {
         Set<Bean<?>> beans = manager.getBeans(name);
 
+        // If it is a Synthetic bean, it should match with type
+        beans = beans.stream()
+                .filter(bean -> {
+                    if (bean instanceof InjectableBean injectableBean) {
+                        return !injectableBean.getKind().equals(InjectableBean.Kind.SYNTHETIC)
+                                || bean.getTypes().contains(type);
+                    } else {
+                        return true;
+                    }
+                }).collect(Collectors.toSet());
+
```

Comment:
@jamesnetherton  @ppalaga I replace to check the beans which are generated by the extensions are `SYNTHETIC`. We don't need to introduce a `NamedAnnotationLiteral` build item. The `EntityManagerFactory` class from `quarkus-hibernete-orm` is also with `@Named` annotation.